service/controller/v24/resource/cpi/create.go

```diff
@@ -0,0 +1,111 @@
+package cpi
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/v24/adapter"
+	"github.com/giantswarm/aws-operator/service/controller/v24/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/v24/key"
+	"github.com/giantswarm/aws-operator/service/controller/v24/templates"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding the tenant cluster's control plane initializer CF stack")
+
+		i := &cloudformation.DescribeStacksInput{
+			StackName: aws.String(key.MainHostPreStackName(customObject)),
+		}
+
+		_, err = r.hostClients.CloudFormation.DescribeStacks(i)
+		if IsNotExists(err) {
+			// fall through
+		} else if err != nil {
+			return microerror.Mask(err)
+		} else {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "found the tenant cluster's control plane initializer CF stack already exists")
+
+			return nil
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the tenant cluster's control plane initializer CF stack")
+	}
+
+	var templateBody string
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "computing the template of the tenant cluster's control plane initializer CF stack")
+
+		cc, err := controllercontext.FromContext(ctx)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		var newAdapter adapter.Adapter
+		{
+			c := adapter.Config{
+				CustomObject:   customObject,
+				GuestAccountID: cc.Status.Cluster.AWSAccount.ID,
+				Route53Enabled: r.route53Enabled,
+			}
+
+			newAdapter, err = adapter.NewHostPre(c)
```

Comment:
Why not `hostPreAdapter`?

Reply:
No string reason. The type is `Adapter`.